Sources/SourceKitLSP/Swift/CodeActions/Demorgan.swift

```diff
@@ -0,0 +1,200 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftSyntax
+import LanguageServerProtocol
+import SwiftRefactor
+
+///
+///
+/// ## Before
+/// ```swift
+///
+/// ```
+///
+/// ## After
+/// ```swift
+///
+/// ```
```

Comment:
Heh the fact I didn't finish documenting this likely means it may not be ready for prime time. I dunno, tho, the implementation looks mostly fine.